debian/control: Lint using cme

rzr opened the pull request:
Change-Id: Ib59e8dd0b489b48b210efb51915b7135695d1438
Fordwarded: https://github.com/theupdateframework/tuf/pulls/rzr
Relate-to: https://github.com/theupdateframework/tuf/issues/263
Signed-off-by: Philippe Coval <rzr@users.sf.net>

Please fill in the fields below to submit a pull request.  The more information
that is provided, the better.

Fixes #<ISSUE NUMBER>

**Description of the changes being introduced by the pull request**:

**Please verify and check that the pull request fulfills the following
requirements**:

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [x] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

rzr committed:
debian/changelog: Update to latest release

Relate-to: https://lists.debian.org/debian-python/2021/03/msg00005.html
Change-Id: Idc7f396a6dd2447c35f82d0a63641e1d4c203c61
Signed-off-by: Philippe Coval <rzr@users.sf.net>

rzr committed:
d/control: Update Maintainer to DPT

Change-Id: I4f19349f135bff1fce3590143900fa7c63d21c5d
Relate-to: https://github.com/theupdateframework/tuf/issues/263
Signed-off-by: Philippe Coval <rzr@users.sf.net>

rzr committed:
d/control: Lint using cme

Change-Id: Ib59e8dd0b489b48b210efb51915b7135695d1438
Fordwarded: https://github.com/theupdateframework/tuf/pull/1300
Relate-to: https://github.com/theupdateframework/tuf/issues/263
Signed-off-by: Philippe Coval <rzr@users.sf.net>

lukpueh commented on debian/control:
```diff
@@ -1,29 +1,26 @@
+               python3-all,
+               python3-setuptools,
+               python3-six,
+               python3-iso8601,
```
iso8601 dependency was removed in v0.15.0. Please remove here too.

rzr replied:
ok will do in later PR

lukpueh commented on debian/control:
```diff
@@ -1,29 +1,26 @@
 Source: python-tuf
+Maintainer: tuf developers <theupdateframework@googlegroups.com>
+Uploaders: Lukas Puehringer <lukas.puehringer@nyu.edu>,
+           Justin Cappos <jcappos@nyu.edu>
```
@rzr, you should probably add your name here?

rzr replied:
done in next change

lukpueh commented on debian/control:
```diff
@@ -1,29 +1,26 @@
+               python3-six,
+               python3-iso8601,
+               python3-requests,
+               python3-securesystemslib
```
Have you tried building the package with tests yet? I think we need to include securesystemslib's optional dependency at least as build dependency, i.e.:
```
 python3-cryptography,
 python3-nacl,
```

rzr replied:
yes I managed to run tests but they don't work offline yet so I disabled then for now, I'll try to enable them once package lands in repo

rzr replied:
yes, I have tests running and even with autopkgtest \o/

I will open a new PR for this, meanwhile If curious you can run them from:

https://salsa.debian.org/rzr/python-tuf

lukpueh reviewed (commented):
Thanks for the PR, @rzr!

rzr commented:
Thanks for the remarks can we  take them into account in later patches (from my debian/master branch)
I don't want to mix lint and functional changes
Thanks

rzr committed:
d/control: Drop python3-iso8601 depencency

It was removed since v0.15.0

Thanks-to: <Lukas Puehringer <lukas.puehringer@nyu.edu>
Relate-to: https://github.com/theupdateframework/tuf/issues/263
Signed-off-by: Philippe Coval <rzr@users.sf.net>

joshuagl commented:
Hi @rzr. I really appreciate your interest in the project, but I'm not sure it makes sense to spend significant effort packaging tuf for Debian at this time.

We're embarking on a [refactoring effort](https://github.com/theupdateframework/tuf/projects/2) and hoping to make a 1.0 release in the foreseeable future, so any package we create today is unlikely to be useful, or supported, for a Debian release cycle and will likely to need overhauling once the refactoring effort is complete.

If you are interested in the project outside of Debian packaging, we'd be happy to help you find some reasonable issues to work on. Otherwise we would be very grateful for your packaging efforts once the refactor has started to take shape and we have a good idea of what to package.

lukpueh reviewed (approved):
Thanks for you contribution, @rzr!

This is an enhancement to the `debian` branch (PR target), so I will go ahead and merge.

To finalize this work the `debian` branch will need to be rebased on `develop`, which has seen lots of development since this PR was opened (see @joshuagl's refactoring comment above).

This packaging effort will continue to be tracked in #263.

rzr commented:
ok thx I'll push more changes that could be also useful in current debian branch, please hold rebase a couple of days then you can resume where I was
